src/common/utils/log/logger.go

```diff
@@ -79,6 +98,58 @@ func DefaultLogger() *Logger {
 	return logger
 }
 
+func (l *Logger) clone() *Logger {
+	return &Logger{
+		out:       l.out,
+		fmtter:    l.fmtter,
+		lvl:       l.lvl,
+		callDepth: l.callDepth,
+		skipLine:  l.skipLine,
+		fields:    l.fields,
+		fieldsStr: l.fieldsStr,
+		mu:        l.mu,
+	}
+}
+
+// WithDepth ...
+func (l *Logger) WithDepth(depth int) *Logger {
+	r := l.clone()
+	r.callDepth = depth
+
+	return r
+}
+
+// WithFields ...
+func (l *Logger) WithFields(fields Fields) *Logger {
+	r := l.clone()
+
+	if len(fields) > 0 {
+		copyFields := make(map[string]interface{}, len(l.fields)+len(fields))
+		for key, value := range l.fields {
+			copyFields[key] = value
+		}
+		for key, value := range fields {
+			copyFields[key] = value
+		}
+
+		sortedKeys := make([]string, 0, len(copyFields))
+		for key := range copyFields {
+			sortedKeys = append(sortedKeys, key)
+		}
+		sort.Strings(sortedKeys)
```

Comment:
After sorting the fields will not be the same sequence with the original one, is this expected?
FuncC(with field c)->funcB(with filed b)->FuncA(with filed a), the log will be: [a,b,c]xxx, right?

Reply:
1. In Golang, iterations over maps are random, sorting here can make the fileds output is ordered in log

2. `FuncC(with field c)->funcB(with filed b)->FuncA(with filed a), the log will be: [a,b,c]xxx, right?` No, the log will be [c]xxx in `FuncC`, [b]xxx in `funcB` and [a]xxx in `FuncA`. `log.G(ctx)` returns the logger saved in the ctx (if not found, returns the default one), and `logger.WithFields` does not modify the logger itself, it returns new logger.